jumanji/environments/packing/job_shop/scenario_generator.py

```diff
@@ -0,0 +1,178 @@
+
+        Args:
+            max_num_jobs: Maximum number of jobs that can be scheduled.
+            max_num_ops: Maximum number of operations per job.
+            max_op_duration: Maximum duration of any operation.
+        """
+
+        self.max_num_jobs = max_num_jobs
+        self.max_num_ops = max_num_ops
+        self.max_op_duration = max_op_duration
+
+    @abc.abstractmethod
+    def __call__(self, key: chex.PRNGKey, num_jobs: int, num_machines: int) -> Scenario:
+        """Call method responsible for generating a new scenario.
+
+        Args:
+            key: jax random key in case stochasticity is used in the instance generation process.
+            num_jobs: Number of jobs in the instance. It has to be between 1 and self.max_num_jobs.
+            num_machines: Number of machines in the instance.
+
+        Returns:
+            A 'Scenario' object, containing:
+              - ops_machine_ids: Array (max_num_jobs, max_num_ops) indicating the machine for
+                each operation, with -1 indicating that the operation does not exist.
+              - ops_durations: Array (max_num_jobs, max_num_ops) indicating the duration of
+                each operation, with -1 indicating that the operation does not exist.
+              - num_ops_per_job: Array (max_num_jobs) indicating number of operations per job,
+                with 0 indicating that the job does not exist.
+              - key: jax random key used to generate the scenario.
+        """
+
+
+class ToyScenarioGenerator(ScenarioGenerator):
+    """`ScenarioGenerator` that can be used as an example. It deterministically outputs a hardcoded
+    instance with 5 jobs, 4 machines, a max of 4 ops for any job, and max duration of 4 time
+    steps for any operation. By construction, this generator has a known, optimal makespan
+    of 8 time steps.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(max_num_jobs=5, max_num_ops=4, max_op_duration=4)
+
+    def __call__(self, key: chex.PRNGKey, num_jobs: int, num_machines: int) -> Scenario:
+        del key
+        del num_jobs
+        del num_machines
+
+        ops_machine_ids = jnp.array(
+            [
+                [2, 3, 1, 2],
+                [3, 2, 0, -1],
+                [1, 3, -1, -1],
+                [0, 3, 0, 0],
+                [1, 0, 1, -1],
+            ],
+            jnp.int32,
+        )
+        ops_durations = jnp.array(
+            [
+                [2, 2, 1, 2],
+                [2, 4, 1, -1],
+                [2, 3, -1, -1],
+                [4, 1, 1, 1],
+                [3, 1, 2, -1],
+            ],
+            jnp.int32,
+        )
+
+        scenario = Scenario(
+            num_jobs=5,
+            num_machines=4,
+            max_num_jobs=5,
+            max_num_ops=4,
+            ops_machine_ids=ops_machine_ids,
+            ops_durations=ops_durations,
+            num_ops_per_job=jnp.array([4, 4, 2, 4, 3], jnp.int32),
+            key=jax.random.PRNGKey(0),
```

Comment:
I know this is whats in the original code, but it seems weird that they generate ta new key here no ? 
Should this just be the key that entered ?